.github/scripts/userbenchmark/aicluster.py

```diff
@@ -0,0 +1,200 @@
+                                and S3Client.get_filename_from_key(x).startswith("metrics-") \
+                                and x.endswith(".json"), \
+                                s3.list_directory(directory=None)))
+        return filtered_metrics
+    def update_index_from_metrics(index, metric_files):
+        metric_filenames = list(map(lambda x: S3Client.get_filename_from_key(x), metric_files))
+        for metric_filename in metric_filenames:
+            if not metric_filename in index:
+                index[metric_filename] = {}
+                index[metric_filename]["uploaded-scribe"] = False
+        return index
+    index_key = s3.exists(prefix=benchmark_name, file_name=INDEX_FILE_NAME)
+    index = gen_index_obj(index_key)
+    metric_files = filter_metric_files(s3.list_directory(directory=None))
+    updated_index = update_index_from_metrics(index, metric_files)
+    return updated_index
+
+def upload_metrics_to_scribe(s3, benchmark_name, index, work_dir):
+    """
+    for each 'uploaded-scrbe: False' file in index
+      1. download it from S3
+      2. upload it to scribe
+      3. if success, update the index file with 'uploaded-scribe: True'
+    upload the updated index file to S3 after processing all files
+    """
+    try:
+        for index_key in index:
+            assert "uploaded-scribe" in index[index_key], \
+                f"Index key {index_key} missing field uploaded-scribe!"
+        index_file_path = work_dir.joinpath(INDEX_FILE_NAME)
+        with open(index_file_path, "w") as index_file:
+            yaml.safe_dump(index, index_file)
+        need_upload_metrics = filter(lambda x: not index[x]["uploaded-scribe"], index.keys())
+        for upload_metrics in need_upload_metrics:
+            # download the metrics file from S3 to work_dir
+            print(f"Downloading metrics file {upload_metrics} to local.")
+            metrics_key = s3.exists(prefix=None, file_name=upload_metrics)
+            assert metrics_key, f"Expected metrics file {upload_metrics} does not exist."
+            s3.download_file(metrics_key, work_dir)
+            # upload it to scribe
+            print(f"Uploading metrics file {upload_metrics} to scribe.")
+            metrics_path = str(work_dir.joinpath(upload_metrics).resolve())
+            with open(metrics_path, "r") as mp:
+                benchmark_time, benchmark_data = process_benchmark_json(mp)
+            uploader = TorchBenchUserbenchmarkUploader()
+            # user who run the benchmark on ai cluster
+            uploader.UNIX_USER = "diegosarina"
+            uploader.SUBMISSION_GROUP_GUID = "ai-cluster"
+            uploader.post_userbenchmark_results(benchmark_time, benchmark_data)
+            # update the index file
+            index[upload_metrics]["uploaded-scribe"] = True
+            with open(index_file_path, "w") as index_file:
+                yaml.safe_dump(index, index_file)
```

Comment:
Do we need to clean up obsolete benchmark results, or have we decided to keep everything?

Reply:
Currently, we keep everything in the Amazon S3 bucket. This historical data can be useful for diagnosis.